src/main/java/net/greatstart/controllers/InvestmentController.java

```diff
@@ -1,131 +1,72 @@
-    public Investment getInvestmentById(@PathVariable long id) {
-        return investmentService.getInvestmentById(id);
-    }
-
-    @GetMapping("/investment")
-    public ModelAndView getAllInvestments() {
-        ModelAndView model = new ModelAndView(INVESTMENTS_VIEW);
-        model.addObject(PAGE_NAME, "All investments.");
-        model.addObject(INVESTMENT_LIST, investmentService.getAllInvestments());
-        return model;
-    }
+    @GetMapping("{id}")
+    public DtoInvestment getInvestmentById(@PathVariable long id) {
 
-    @GetMapping("/project/{id}/investments")
-    public ModelAndView getAllProjectInvestments(@PathVariable long id) {
-        ModelAndView model = new ModelAndView(INVESTMENTS_VIEW);
-
-        model.addObject(PAGE_NAME,
-                "Investments in project: "
-                        + projectService.getProjectById(id).getDesc().getName());
-        model.addObject(INVESTMENT_LIST,
-                projectService.getProjectById(id).getInvestments());
-        return model;
-    }
-
-    @GetMapping("/user/investments")
-    public ModelAndView getAllUserInvestments(Principal principal) {
-        ModelAndView model = new ModelAndView(INVESTMENTS_VIEW);
-        model.addObject(PAGE_NAME, String.format("Investments of user: %s %s",
-                userService.getUserByEmail(principal.getName()).getName(),
-                userService.getUserByEmail(principal.getName()).getLastName()));
-        model.addObject(INVESTMENT_LIST,
-                userService.getUserByEmail(principal.getName()).getInvestments());
-        return model;
+        return investmentService.getDtoInvestmentById(id);
     }
 
-
-    @GetMapping("/investment/{id}/delete")
-    public ModelAndView deleteInvestmentById(@PathVariable long id) {
-        investmentService.deleteInvestment(id);
-        return new ModelAndView(REDIRECT + INVESTMENTS_PAGE);
+    @PostMapping
+    public ResponseEntity<DtoInvestment> createInvestment(@Valid @RequestBody DtoInvestment investment) {
+        investment.setDateOfInvestment(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS));
+        investment.setVerified(false);
+        investment.setVerified(false);
+        DtoInvestment investmentResult = investmentService.saveInvestment(investment);
+        if (investmentResult != null && investmentResult.getId() != 0) {
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.EXPECTATION_FAILED);
     }
 
-    private BigDecimal getInvestedSumInProject(long id) {
-        return projectService.getProjectById(id).getInvestments()
-                .stream().map(Investment::getSum).reduce(BigDecimal.ZERO, BigDecimal::add);
+    @PutMapping("{id}")
+    public ResponseEntity<DtoInvestment> updateInvestment(@PathVariable long id,
+                                                          @RequestBody DtoInvestment investment) {
+        //todo: implement update investment
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)